src/main/java/org/elasticsearch/cluster/ClusterState.java

```diff
@@ -249,6 +253,10 @@ public ClusterBlocks getBlocks() {
         return this.customs;
     }
 
+    public <T extends Custom> T custom(String type) {
+        return (T) customs.get(type);
```

Comment:
`customPrototype` contains a list of dummy (prototype) objects that are used for deserialization when we don't have a concrete implementation to diff against. It is static and changes only when a component needs to register another custom object. In other words you can think of `customPrototypes` as factories. The `custom` map contains the actual objects that are part of the cluster state. 

We could use `.customs().get("mycustom")` but it would return us objects with type `Custom` that we would have to later cast to concrete custom types. This is a convenience method that does this cast for us, so later we can write something like:
```
SnapshotsInProgress snapshotsInProgress = allocation.routingNodes().custom(SnapshotsInProgress.TYPE);
```
instead of
```
SnapshotsInProgress snapshotsInProgress = (SnapshotsInProgress) allocation.routingNodes().custom().get(SnapshotsInProgress.TYPE);
```